internal/datanode/services.go

```diff
@@ -271,8 +280,62 @@ func (node *DataNode) SyncSegments(ctx context.Context, req *datapb.SyncSegments
 		return merr.Status(err), nil
 	}
 
-	// TODO: sheep, add a new DropCompaction interface, deprecate SyncSegments
-	node.compactionExecutor.removeTask(req.GetPlanID())
+	if len(req.GetSegmentInfos()) <= 0 {
+		log.Info("sync segments is empty, skip it")
+		return merr.Success(), nil
+	}
+
+	ds, ok := node.flowgraphManager.GetFlowgraphService(req.GetChannelName())
+	if !ok {
+		node.compactionExecutor.discardPlan(req.GetChannelName())
+		err := merr.WrapErrChannelNotFound(req.GetChannelName())
+		log.Warn("failed to get flow graph service", zap.Error(err))
+		return merr.Status(err), nil
+	}
+
+	allSegments := make(map[int64]struct{})
+	for segID := range req.GetSegmentInfos() {
+		allSegments[segID] = struct{}{}
+	}
+
+	missingSegments := ds.metacache.DetectMissingSegments(allSegments)
+
+	newSegments := make([]*datapb.SyncSegmentInfo, 0, len(missingSegments))
+	futures := make([]*conc.Future[any], 0, len(missingSegments))
+
+	for _, segID := range missingSegments {
```

Comment:
The timeout for the SyncSegments RPC is 10 seconds. If there are many segments, it may time out due to loading 
 many stats log.

Reply:
will fix it in next pr, master also need it.